### ISSUE TITLE
`Router::route_service()` example code is out of date

### ISSUE THREAD

**Issue by jwodder** [tokio-rs/axum#2536]
[The current documentation for `axum::Router::route_service()`](https://docs.rs/axum/0.7.4/axum/struct.Router.html#method.route_service) has an example that contains the following lines:

```
        // Services whose response body is not `axum::body::BoxBody`
        // can be wrapped in `axum::routing::any_service` (or one of the other routing filters)
        // to have the response body mapped
        any_service(service_fn(|_: Request| async {
            let res = Response::new(Body::from("Hi from `GET /`"));
            Ok::<_, Infallible>(res)
        }))

        ...

        // This service's response body is `axum::body::BoxBody` so
        // it can be routed to directly.
        service_fn(|req: Request| async move {
            let body = Body::from(format!("Hi from `{} /foo`", req.method()));
            let res = Response::new(body);
            Ok::<_, Infallible>(res)
        })
```

However, despite what the comments in the code say, both responses appear to have `axum::body::Body` as their bodies, and there is currently no `axum::body::BoxBody` in the crate.  I assume that `BoxBody` was removed in an earlier version and the example was only partially updated, but that just raises the question: What is `any_service()` good for now?

**Comment by davidpdrsn** [comment]
Ah yes that example is indeed out of date. axum always uses `axum::body::Body` for requests and responses since 0.7.

> What is any_service() good for now?

It can still be used to route to any service regardless of the HTTP method.

**Comment by jwodder** [comment]
> > What is any_service() good for now?
> 
> It can still be used to route to any service regardless of the HTTP method.

So what happens if I supply a `service_fn()` but don't wrap it in `any_service()`?  Will only GET requests be routed to it?

**Comment by davidpdrsn** [comment]
"Supply" it how?

**Comment by jwodder** [comment]
Ah, I missed that the non-`any_service()` bit of the example was using `route_service()` instead of `route()`.  So the main point of `any_service()` is to let you pass a service to `route()` instead of to `route_service()`?

**Comment by davidpdrsn** [comment]
Exactly. You might also wanna run the MethodRouter, without using Router at all.

**Comment by JosiahParry** [comment]
It would be great if the example code or documentation could illustrate how to use this with State and Path. 

I have a function like so
```rust
async fn handler(State(db): State<Arc<Database>>,
    Path(id): Path<String>,
) 
```

and providing to `route_service("/:id", service_fn(handler))` results in a compiler error.

There isn't any documentation or examples to figure out how to resolve this unfortunately

**Comment by jplatte** [comment]
@JosiahParry `service_fn` does not support axum handler functions, it is much more restrictive. Please can you open a separate question in the [Q&A section](https://github.com/tokio-rs/axum/discussions/categories/q-a)?

**Comment by wyf027** [comment]
Combined documentation PR: #3767 (supersedes #3766).

**Comment by KayanoLiam** [comment]
Hi, I’ll have a go at sorting this out. I’ll update the `route_service()` example to match the current `Body` API and make it clearer when `any_service()` is useful. I’ll check the docs and open a PR once it’s ready.
